test/Microsoft.ML.Tests/Transformers/KeyToVectorEstimatorTests.cs

```diff
@@ -222,5 +223,50 @@ public void TestOldSavingAndLoading()
                 var loadedView = ModelFileUtils.LoadTransforms(Env, dataView, ms);
             }
         }
+
+        public class ModelInput
+        {
+            [ColumnName("Label"), LoadColumn(0)]
+            public int Label { get; set; }
+
+
+            [ColumnName("ProblematicColumn"), LoadColumn(1)]
+            public string ProblematicColumn { get; set; }
+        }
+
+        static IEnumerable<ModelInput> GetData()
+        {
+            for (int i = 0; i < 1000; i++)
+            {
+                yield return new ModelInput { Label = i % 3, ProblematicColumn = (i % 200).ToString() };
+            }
+        }
+
+        [Fact]
+        public void KeyToVectorOverflowTest()
+        {
+            // This test is introduced for https://github.com/dotnet/machinelearning/issues/5211
+            // that provides users an informational exception message
+            MLContext mlContext = new MLContext(1);
+
+            IDataView dataview = mlContext.Data.LoadFromEnumerable(GetData());
+
+            var pipeline = mlContext.Transforms.Conversion.MapValueToKey("Label")
+                   .Append(mlContext.Transforms.Categorical.OneHotHashEncoding("ProblematicColumn"));
+
+            var featurizedData = pipeline.Fit(dataview).Transform(dataview);
+
+            try
+            {
+                var transformer = pipeline.Fit(featurizedData);
+                Assert.True(false);
+            }
+            catch (System.Exception ex)
+            {
```

Comment:
You can catch the `OverflowException` instead here.

[https://docs.microsoft.com/en-us/dotnet/api/system.overflowexception?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.overflowexception?view=netcore-3.1)